rest-service/manager_rest/resource_manager.py

```diff
@@ -78,9 +79,32 @@ def update_execution_status(self, execution_id, status, error):
                 .format(execution.status, status))
         execution.status = status
         execution.error = error
+
         if status in ExecutionState.END_STATES:
             execution.ended_at = utils.get_formatted_timestamp()
-        return self.sm.update(execution)
+
+        res = self.sm.update(execution)
+        if status in ExecutionState.END_STATES:
+            self.start_queued_executions()
+        return res
+
+    def start_queued_executions(self):
+        queued_executions = self._get_queued_executions()
+
+        for e in queued_executions:
+            self.execute_queued_workflow(e)
+            if e.is_system_workflow:  # To avoid starvation of system workflows
+                break
+
+    def _get_queued_executions(self):
+        filters = {'status': ExecutionState.QUEUED_STATE}
+        queued_executions = [
+            e for e in self.list_executions(is_include_system_workflows=True,
+                                            filters=filters,
+                                            sort={'created_at': 'asc'}).items
```

Comment:
can we also sort that system execs come first?

Reply:
@idob-cloudify in the loop calling this, we break on the first system workflow found "to avoid starving system executions"; but it's still possible that there's many non-system execs and the system exec will have to wait

if we sorted this so that system execs come first, then the queueing would prioritize system execs, maybe that's what we want?

Reply:
no, we want to execute in the order of creation.